discovery/marathon/marathon.go

```diff
@@ -201,11 +213,20 @@ type AppList struct {
 }
 
 // AppListClient defines a function that can be used to get an application list from marathon.
-type AppListClient func(client *http.Client, url string) (*AppList, error)
+type AppListClient func(client *http.Client, url, token string) (*AppList, error)
 
 // fetchApps requests a list of applications from a marathon server.
-func fetchApps(client *http.Client, url string) (*AppList, error) {
-	resp, err := client.Get(url)
+func fetchApps(client *http.Client, url, token string) (*AppList, error) {
+	request, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if token != "" {
+		request.Header.Set("Authorization", "token="+token)
+	}
```

Comment:
I see, makes sense then. Would be best to add an appropriate comment so we know why in the future.